Utility.cs

```diff
@@ -114,10 +115,15 @@ public static partial class Utility {
 			return equal;
 		}
 
-		private static byte[] TagIOSave(Item item) {
-			using MemoryStream memoryStream = new();
-			TagIO.ToStream(ItemIO.Save(item), memoryStream);
-			return memoryStream.ToArray();
+		private static string TagIOSave(Item item, ConditionalWeakTable<Item, string> cache) {
+            // if present in cache, return cached value
```

Comment:
accidental spaces instead of tabs?

Reply:
gah, even with the whitespace shown, sometimes I miss it apparently!